src/effects/effectslot.cpp

```diff
@@ -268,3 +272,62 @@ void EffectSlot::slotEffectMetaParameter(double v, bool force) {
         pParameterSlot->onEffectMetaParameterChanged(v, force);
     }
 }
+
+QDomElement EffectSlot::toXml(QDomDocument* doc) const {
+    QDomElement effectElement = doc->createElement(EffectXml::Effect);
+    if (m_pEffect == nullptr) {
+        return effectElement;
+    }
+
+    QDomElement metaKnobElement = doc->createElement(EffectXml::EffectMetaParameter);
+    XmlParse::addElement(*doc, effectElement,
+                         EffectXml::EffectMetaParameter,
+                         QString::number(m_pControlMetaParameter->get()));
+    EffectManifest manifest = m_pEffect->getManifest();
+    XmlParse::addElement(*doc, effectElement,
+                         EffectXml::EffectId, manifest.id());
+    XmlParse::addElement(*doc, effectElement,
+                         EffectXml::EffectVersion, manifest.version());
+
+    QDomElement parametersElement = doc->createElement(EffectXml::ParametersRoot);
+
+    for (const auto& pParameter : m_parametersById) {
+        QDomElement parameterElement = pParameter->toXml(doc);
+        XmlParse::addElement(*doc, parameterElement,
+                             EffectXml::ParameterId,
+                             pParameter->getManifest().id());
+        parametersElement.appendChild(parameterElement);
+    }
+
+    effectElement.appendChild(parametersElement);
+
+    return effectElement;
+}
+
+void EffectSlot::loadEffectSlotFromXml(const QDomElement& effectElement) {
+    if (effectElement.text().isEmpty()) {
+        return;
+    }
+
```

Comment:
Since loading the effect chain into the effect chain slot and loading the chain slot's XML config happen separately, want to add a sanity check here that the XML effect ID matches the loaded effect ID and print a warning if it doesn't?